http/client.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package http
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/spacemonkeygo/monkit/v3"
+)
+
+// TraceRequest will perform an HTTP request, creating a new Span for the HTTP
+// request and sending the Span in the HTTP request headers.
+// Compare to http.Client.Do.
+func TraceRequest(ctx context.Context, scope *monkit.Scope, cl Client, req *http.Request, sampled func(trace *monkit.Trace) bool) (
```

Comment:
what's up with the sampled argument? why would someone want that?

Reply:
Monkit has no idea if the traces are sampled or not. For monkit, everything is recorded. Sampling is related to the collectors. 

For example `monkit-jaeger` [uses](https://github.com/storj/common/blob/3816c1e17d27f58cfd28db45a3d7532ac2c83eb8/rpc/rpctracing/common.go#L17) a constant from `storj/common`. 

I didn't decide if I love it (monkit is uber general) or not (every sampling is different), but this function parameter makes it possible to put this helper to monkit even today: the logic how the sampling is turned off/on can be plugged in.

Reply:
monkit does actually know about sampling, at least enough for the present package to be able to mark a span sampled: https://github.com/spacemonkeygo/monkit/blob/master/present/path.go#L242. does that change your opinion at all?

Reply:
Ah, nice. Didn't find it. Looks to be a new thing, maybe we can switch to use it with `monkit-jaeger`, too.

((Maybe we can move  it out from the `present` package to make it more visible?))

> does that change your opinion at all?

Sure. It changed my opinion and the code, too. Just pushed... ;-)

Reply:
oh sure, we can move it out of present. in particular, i think this code is still confusing to me:

```golang
func TraceInfoFromSpan(s *monkit.Span, sampledf func(trace *monkit.Trace) bool) TraceInfo {
	trace := s.Trace()

	sampled := sampledf(trace)
	if !sampled {
		return TraceInfo{Sampled: sampled}
	}
```

why isn't this 
```golang
func TraceInfoFromSpan(s *monkit.Span) TraceInfo {
	trace := s.Trace()

	sampled, _ := trace.Get(SampledKey).(bool)
	if !sampled {
		return TraceInfo{Sampled: sampled}
	}
```